### ISSUE TITLE
Dart SDK -> Engine roll blocked on web engine build failures

### ISSUE THREAD

**Issue by zanderso** [flutter/flutter#134428]
The roll was stopped for branch alignment for a few days. When restarted here https://github.com/flutter/engine/pull/45581, the web engine failed to build:

https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20Engine%20Drone/1331881/overview

@eyebrowsoffire bisected to the change being reverted here https://dart-review.googlesource.com/c/sdk/+/324902

There is a potential engine-side fix in https://github.com/flutter/engine/pull/45591, but the presub failures look like they might be related to the change.

cc @srujzs @sigmundch

**Comment by eyebrowsoffire** [comment]
I tried to get the engine side fix working on Friday, but ran into some issues with some stuff we're doing in one of our unit tests. I'm not 100% sure what's going on, but I think there are some subtle differences between `allowInterop` and the `toJS` function conversion that don't interact well with some admittedly weird stuff we are doing in our unit tests. I'll try to get a more definitive answer today.

In the meantime I suggest we land the revert on the Dart SDK side ASAP to unblock the roller, and reland it after I am confident we have an engine-side fix that works properly.

**Comment by eyebrowsoffire** [comment]
As an update, I have a PR out on the engine side to prepare things for the reland of the dart changes that make JS interop declarations more strict about what types can be used: https://github.com/flutter/engine/pull/45668

The other failure Zach posted about above seems like a separate issue. I will try to figure out the breaking dart change in a bit.

**Comment by eyebrowsoffire** [comment]
I have a revert that should resolve the remaining issues blocking the dart -> engine roll: https://dart-review.googlesource.com/c/sdk/+/325445

Also, https://github.com/flutter/engine/pull/45668 just landed, and I think we can at least attempt a roll to dart 3.2.0-140.0, which is before the other breaking change.

**Comment by a-siva** [comment]
Can this be closed? Looks like the dart rollers are running now.

**Comment by zanderso** [comment]
Yes.
